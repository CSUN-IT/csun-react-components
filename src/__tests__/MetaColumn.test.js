import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MetaColumn from '../../src/components/meta/Column';

configure({ adapter: new Adapter() });

describe('MetaColumn', () => {
  it('Should be defined', () => {
    expect(MetaColumn).toBeDefined();
  });

  it('Renders component correctly', () => {
    const tree = shallow(
      <MetaColumn sizes={['sm']} values={['4']}>
        1
      </MetaColumn>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders children correctly', () => {
    const tree = mount(
      <div>
        <MetaColumn sizes={['sm']} values={['4']}>
          1
        </MetaColumn>
        <MetaColumn sizes={['sm']} values={['4']}>
          2
        </MetaColumn>
      </div>,
    );
    expect(tree.find(MetaColumn)).toHaveLength(2);
  });
});
