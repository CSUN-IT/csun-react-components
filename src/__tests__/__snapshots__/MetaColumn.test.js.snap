// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaColumn Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaColumn
    className=""
    sizes={
      Array [
        "sm",
      ]
    }
    values={
      Array [
        "4",
      ]
    }
  >
    1
  </MetaColumn>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "1",
      "className": " col-sm-4",
    },
    "ref": null,
    "rendered": "1",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "1",
        "className": " col-sm-4",
      },
      "ref": null,
      "rendered": "1",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
